crates/kilt-dip-primitives/src/merkle/v0.rs

```diff
@@ -0,0 +1,1299 @@
+impl<Context> kilt_support::traits::GetWorstCase<Context> for DipCommitmentStateProof {
+	fn worst_case(context: Context) -> Self {
+		Self(BoundedBlindedValue::worst_case(context))
+	}
+}
+
+/// The Merkle proof for a KILT DID.
+///
+/// The generic types indicate the following:
+/// * `ProviderDidKeyId`: The DID key ID type configured by the provider.
+/// * `ProviderAccountId`: The `AccountId` type configured by the provider.
+/// * `ProviderBlockNumber`: The `BlockNumber` type configured by the provider.
+/// * `ProviderWeb3Name`: The web3name type configured by the provider.
+/// * `ProviderLinkableAccountId`: The linkable account ID type configured by
+///   the provider.
+#[derive(Clone, Debug, Encode, Decode, PartialEq, Eq, TypeInfo)]
+pub struct DidMerkleProof<
+	ProviderDidKeyId,
+	ProviderAccountId,
+	ProviderBlockNumber,
+	ProviderWeb3Name,
+	ProviderLinkableAccountId,
+> {
+	pub(crate) blinded: BoundedBlindedValue<u8>,
+	pub(crate) revealed: Vec<
+		RevealedDidMerkleProofLeaf<
+			ProviderDidKeyId,
+			ProviderAccountId,
+			ProviderBlockNumber,
+			ProviderWeb3Name,
+			ProviderLinkableAccountId,
+		>,
+	>,
+}
+
+impl<ProviderDidKeyId, ProviderAccountId, ProviderBlockNumber, ProviderWeb3Name, ProviderLinkableAccountId>
+	DidMerkleProof<ProviderDidKeyId, ProviderAccountId, ProviderBlockNumber, ProviderWeb3Name, ProviderLinkableAccountId>
+{
+	pub fn new(
+		blinded: BoundedBlindedValue<u8>,
+		revealed: Vec<
+			RevealedDidMerkleProofLeaf<
+				ProviderDidKeyId,
+				ProviderAccountId,
+				ProviderBlockNumber,
+				ProviderWeb3Name,
+				ProviderLinkableAccountId,
+			>,
+		>,
+	) -> Self {
+		Self { blinded, revealed }
+	}
+}
+
+#[cfg(feature = "runtime-benchmarks")]
+impl<
+		ProviderDidKeyId,
+		ProviderAccountId,
+		ProviderBlockNumber,
+		ProviderWeb3Name,
+		ProviderLinkableAccountId,
+		Context,
+	> kilt_support::traits::GetWorstCase<Context>
+	for DidMerkleProof<
+		ProviderDidKeyId,
+		ProviderAccountId,
+		ProviderBlockNumber,
+		ProviderWeb3Name,
+		ProviderLinkableAccountId,
+	> where
+	ProviderDidKeyId: Default + Clone,
+	ProviderAccountId: Clone,
+	ProviderBlockNumber: Default + Clone,
+	ProviderWeb3Name: Clone,
+	ProviderLinkableAccountId: Clone,
+{
+	fn worst_case(context: Context) -> Self {
+		Self {
+			blinded: BoundedBlindedValue::worst_case(context),
+			revealed: sp_std::vec![RevealedDidMerkleProofLeaf::default(); 64],
+		}
+	}
+}
+
+/// A DID signature anchored to a specific block height.
+///
+/// The generic types indicate the following:
+/// * `BlockNumber`: The `BlockNumber` definition of the chain consuming (i.e.,
+///   validating) this signature.
+#[derive(Clone, Debug, Encode, Decode, PartialEq, Eq, TypeInfo)]
+pub struct TimeBoundDidSignature<BlockNumber> {
+	/// The signature.
+	pub(crate) signature: DidSignature,
+	/// The block number until the signature is to be considered valid.
+	pub(crate) valid_until: BlockNumber,
+}
+
+#[cfg(feature = "runtime-benchmarks")]
```

Comment:
Maybe it is worth it to move all mocked structs required for benchmarks into a file?

Reply:
This will go entirely with the upcoming benchmarking PR. I already have the changes ready, but will put them in a self-contained PR. Good point tho.